plugins/module_utils/network/sonic/config/dhcp_snooping/dhcp_snooping.py

```diff
@@ -0,0 +1,636 @@
+        input expected as a dictionary of form {"afi": <ip_version>, "vlans": <list_of_vlans>}'''
+        requests = []
+        if afi.get('vlans'):
+            for vlan_id in afi.get('vlans'):
+                requests.append({
+                    'path': self.vlans_uri.format(vlan_name='Vlan' + vlan_id, v=self.afi_to_vnum(afi)),
+                    'method': DELETE
+                })
+        return requests
+
+    def get_delete_trusted_requests(self, afi):
+        '''makes and returns request to delete the given trusted interfaces for the given afi faimily.
+        input expected as a dictionary of form {"afi": <ip_version>, "trusted": [{"intf_name": <name>}...]}'''
+        requests = []
+        if afi.get('trusted'):
+            for intf in afi.get('trusted'):
+                intf_name = intf.get('intf_name')
+                if intf_name:
+                    requests.append({
+                        'path': self.trusted_uri.format(name=intf_name, v=self.afi_to_vnum(afi)),
+                        'method': DELETE
+                    })
+        return requests
+
+    def get_delete_all_source_bindings_request(self):
+        '''creates request to delete the source bindings list'''
+        return [{'path': self.binding_uri, 'method': DELETE}]
+
+    def get_delete_specific_source_bindings_requests(self, afi):
+        '''creates and builds a list of requests to delete the source bindings listed in the given afi family
+        input expected as a dictionary of form to_delete = {"afi": <ip_version>, "source_bindings": <list of source_bindings>}'''
+        requests = []
+        for entry in afi.get('source_bindings'):
+            if entry.get('mac_addr'):
+                requests.append({
+                    'path': self.binding_uri + '={mac},{ipv}'.format(mac=entry.get('mac_addr'), ipv=afi.get('afi')),
+                    'method': DELETE
+                })
+        return requests
+
+    def get_delete_individual_source_bindings_requests(self, afi, entry):
+        '''create a request to delete the given source binding entry and address family specified
+        by afi'''
+        return [{'path': self.binding_uri + '={mac},{ipv}'.format(mac=entry.get('mac_addr'), ipv=afi.get('afi')), 'method': DELETE}]
+
+    def get_delete_replaced_groupings(self, afis):
+        '''builds list of requests to handle replaced state for both address families'''
+        modified_afi_commands = []
+        requests = []
+
+        want_ipv4 = afis.get('want_ipv4')
+        have_ipv4 = afis.get('have_ipv4')
+        want_ipv6 = afis.get('want_ipv6')
+        have_ipv6 = afis.get('have_ipv6')
+
+        if want_ipv4 and have_ipv4:
+            ipv4_commands, ipv4_requests = self.get_delete_replaced_groupings_afi(want_ipv4, have_ipv4)
+            requests.extend(ipv4_requests)
+            if ipv4_commands:
+                ipv4_commands["afi"] = deepcopy(want_ipv4["afi"])
```

Comment:
Since `afi` is a str option, deepcopy is not required.
```suggestion
                ipv4_commands["afi"] = want_ipv4["afi"]
```